src/IISIntegration/test/WebSites/OutOfProcessWebSite/Startup.cs

```diff
@@ -97,9 +90,10 @@ private async Task ProcessId(HttpContext context)
             await context.Response.WriteAsync(Process.GetCurrentProcess().Id.ToString());
         }
 
-        private async Task ServerAddresses(HttpContext context)
+        private async Task OriginalServerAddresses(HttpContext ctx)
         {
-            await context.Response.WriteAsync(string.Join(",", _serverAddresses.Addresses));
+            var serverAddresses = ctx.RequestServices.GetService<OriginalServerAddressesFilter>().ServerAddresses;
```

Comment:
This is exactly why out-of-proc can't replace IServerAddressesFeature, it creates ambiguity about which addresses you're asking for.